### PR TITLE
api: add section on gateway

### PR DESCRIPTION
**Pull request by jzelinskie**
This is not a fully fledged document on how to configure this functionality for SpiceDB, but rather an overview for those learning about the capabilities of the API.

### CODE REVIEW

File: docs/reference/api.md [hunk 1]
```diff
@@ -9,6 +9,28 @@ The API can be accessed by [various clients] from programming languages or the c
...
+
+Authzed users can find their API Token in the [Authzed dashboard].
+
+[Bearer Token]: https://datatracker.ietf.org/doc/html/rfc6750#section-2.1
+[gRPC Authentication documentation]: https://grpc.io/docs/guides/auth
+[Authzed dashboard]: https://app.authzed.com
+
+## Alternative HTTP/JSON API
+
+By configuring an additional port to listen for traffic, SpiceDB can also support handling JSON requests over HTTP.
```
Comment (samkim) [inline docs/reference/api.md:26]: The "additional port" part is a bit vague. Could mention the specific `http` flag or just to look at spicedb's available flags. Something actionable for someone who wants to run this.

Comment (jzelinskie) [reply docs/reference/api.md:26]: This isn't a doc for configuring SpiceDB so I do not want to cover CLI flags here.

**Review by samkim — APPROVED** [review]
LGTM

File: docs/reference/api.md (hunk 1 above)
Comment (jzelinskie) [reply docs/reference/api.md:26]: In the future, we can link to that doc to make it actionable.

**Pull request merged**